tests/unit/lms/services/assignment_test.py

```diff
@@ -1,51 +1,138 @@
+            svc.get("TOOL_CONSUMER_INSTANCE_GUID", "RESOURCE_LINK_ID")
+
+    def test_get_returns_None_if_theres_no_matching_assignment_with_two_ids(self, svc):
+        with pytest.raises(NoResultFound):
+            svc.get(
+                "TOOL_CONSUMER_INSTANCE_GUID",
+                "RESOURCE_LINK_ID",
+                "EXT_LTI_ASSIGNMENT_ID",
+            )
+
+    def test_get_by_resource_link_id_only(self, svc, assignment):
         retrieved_assignment = svc.get(
             assignment.tool_consumer_instance_guid, assignment.resource_link_id
         )
 
         assert retrieved_assignment == assignment
 
-    def test_get_returns_None_if_theres_no_matching_assignment(self, svc):
+    def test_get_by_ext_lti_id_only(self, svc, assignment_canvas_not_launched):
         retrieved_assignment = svc.get(
-            "TOOL_CONSUMER_INSTANCE_GUID", "RESOURCE_LINK_ID"
+            assignment_canvas_not_launched.tool_consumer_instance_guid,
+            ext_lti_assignment_id=assignment_canvas_not_launched.ext_lti_assignment_id,
         )
 
-        assert retrieved_assignment is None
+        assert retrieved_assignment == assignment_canvas_not_launched
 
-    def test_get_document_url_returns_the_document_url(self, svc, assignment):
-        document_url = svc.get_document_url(
-            assignment.tool_consumer_instance_guid, assignment.resource_link_id
+    def test_get_by_both_ids_not_launched(self, svc, assignment_canvas_not_launched):
+        assert assignment_canvas_not_launched.resource_link_id is None
+
+        retrieved_assignment = svc.get(
+            assignment_canvas_not_launched.tool_consumer_instance_guid,
+            resource_link_id="RESOURCE_LINK_ID",
+            ext_lti_assignment_id=assignment_canvas_not_launched.ext_lti_assignment_id,
         )
 
-        assert document_url == assignment.document_url
+        assert retrieved_assignment == assignment_canvas_not_launched
+        assert assignment_canvas_not_launched.resource_link_id == "RESOURCE_LINK_ID"
 
-    def test_get_document_url_returns_None_if_theres_no_matching_document_url(
-        self, svc
-    ):
-        document_url = svc.get_document_url(
-            "TOOL_CONSUMER_INSTANCE_GUID", "RESOURCE_LINK_ID"
+    def test_get_by_both_ids_launched(self, svc, assignment_canvas):
+        retrieved_assignment = svc.get(
+            assignment_canvas.tool_consumer_instance_guid,
+            resource_link_id=assignment_canvas.resource_link_id,
+            ext_lti_assignment_id=assignment_canvas.ext_lti_assignment_id,
+        )
+        assert retrieved_assignment == assignment_canvas
+
+    @pytest.mark.parametrize(
+        "old_extra,new_extra",
+        [
+            ({}, {}),
+            ({"canvas_file_mappings": {1: 2}}, {"canvas_file_mappings": {1: 2}}),
+        ],
+    )
+    def test_get_merge_existing(
+        self, old_extra, new_extra, svc, assignment, assignment_canvas_not_launched
+    ):  # pylint:disable=protected-access
```

Comment:
You can use the `db_session` fixture instead of `svc._db`, then you won't need the `# pylint:disable=protected-access`